### DISCUSSION TITLE
What developer tools you are using?

### DISCUSSION THREAD

**Discussion by return42** [searx/searx#2430] [Development]
Just for interest I want to start a discussion about the developer tools used by the members of the searx community.  We all have our favourites. E.g. I like to emacs & linux based toolchains, but I also know that these are not the right tools for everyone and I often struggle when newcomers ask me which tool they should use to be more productive.

If someone is interested in how I integrate emacs, git and linux commands in my searx development cycles, I can write down some words about. But first let me here what you prefer :)

**Comment by virtadpt** [comment]
Linux + gvim or vim + git + Screen

**Comment by dalf** [comment]
I use:
* Ubuntu
* liquidprompt / [bat](https://github.com/sharkdp/bat)
* vscode with various extensions: pylint, better jinja, git history, magicpython, mypy, github, github pull requets and issues and some others.
* pyenv to switch between the different python versions.
* nvm to switch between the different node versions.

Perhaps we can create different sticky threads in the https://github.com/searx/searx/discussions/categories/development : one thread per editor (vim, emacs, vscode, pycharm).

As an alternative, we can a "developer tools" section in the documentation but it is not more about the searx.
